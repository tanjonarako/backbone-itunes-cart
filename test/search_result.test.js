import Backbone from "backbone";
import SearchResultView from "../src/views/search_result";
import SearchResultCollection from "../src/models/songs";

describe('views:SearchResult', function() {
    const list = new SearchResultCollection([
        {trackId: '1', trackName: 'Shape of you', artistName: 'Ed Sheeran'},
        {trackId: '2', trackName: 'Perfect', artistName: 'Ed Sheeran'}
    ]);

    it('should render a list', function() {
        const listSongsView = new SearchResultView({collection: list}).render();

        expect(listSongsView.$el.html()).toMatchSnapshot();
    });

    it('should call add to cart', function() {
        const spy = jest.fn();
        const listSongsView = new SearchResultView({collection: list, handleAddToCart: spy}).render();

        listSongsView.$el.find('.js-btn-add[data-id=1]').click();

        expect(spy).toBeCalledWith('1');
    });
});
