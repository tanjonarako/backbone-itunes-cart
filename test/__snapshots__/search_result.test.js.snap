// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views:SearchResult should render a list 1`] = `
"
    <ul class=\\"list-unstyled\\">
    
        <li class=\\"song-list__song row-flex p-3\\">
            <img src=\\"\\" width=\\"60\\" height=\\"60\\">
            <div class=\\"song-list__song-details pl-3 col-flex\\">
                <h2 class=\\"song-list__song-name m-0\\">Shape of you</h2>
                <span class=\\"song-list__song-artist m-0\\">Ed Sheeran</span>
            </div>
            <button type=\\"button\\" data-id=\\"1\\" class=\\"js-btn-add song-list__add-to-cart\\"></button></li>
        
        <li class=\\"song-list__song row-flex p-3\\">
            <img src=\\"\\" width=\\"60\\" height=\\"60\\">
            <div class=\\"song-list__song-details pl-3 col-flex\\">
                <h2 class=\\"song-list__song-name m-0\\">Perfect</h2>
                <span class=\\"song-list__song-artist m-0\\">Ed Sheeran</span>
            </div>
            <button type=\\"button\\" data-id=\\"2\\" class=\\"js-btn-add song-list__add-to-cart\\"></button></li>
        
    </ul>

"
`;
